### PR TITLE
Release 6.0.0

### PR DESCRIPTION
**Pull request by bruno-garcia**
Draft PR so we can keep up with 6.0.0 and `main` diff.

The idea is to merge this branch into `main` once we GA.

_#skip-changelog_

- [ ] Document the interface for custom serialization for: https://docs.sentry.io/platforms/java/enriching-events/context/

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,29 @@
 # Changelog
 
+## 6.x.x
```
Comment (bruno-garcia) [inline CHANGELOG.md:3]: Is there a reason this won't be 6.0.0?

```suggestion
## 6.0.0
```

**Comment by marandaneto** [comment]
https://github.com/getsentry/sentry-java/pull/1832 has to be implemented in this PR now, due to adding a new field to the device protocol.

**Comment by romtsn** [comment]
@marandaneto you want me to review 17k+ loc additions? I think I will fail miserably at that 😅  But I can give a quick look maybe

**Comment by marandaneto** [comment]
> @marandaneto you want me to review 17k+ loc additions? I think I will fail miserably at that 😅 But I can give a quick look maybe

just a draft for now, just added you to be aware of what's going on.

**Review by marandaneto — APPROVED** [review]

**Pull request merged**